### PR TITLE
Add repository forks and pull requests from forks to FakeGitHub imple…

### PR DESCRIPTION
**Pull request by daschult**
…mentation

### CODE REVIEW

File: lib/github.ts [hunk 1]
```diff
@@ -651,6 +659,43 @@ export class FakeGitHub implements GitHub {
     return result;
   }
 
+  public forkRepository(repository: string | GitHubRepository, forkedRepositoryOwner: string): Promise<FakeGitHubRepository> {
+    return this.getRepository(repository)
+      .then((fakeRepository: FakeGitHubRepository) => {
```
Comment (kpajdzik) [inline lib/github.ts]: Why then instead of awaiting?

File: lib/github.ts [hunk 2]
```diff
@@ -651,6 +659,43 @@ export class FakeGitHub implements GitHub {
...
+        let result: Promise<FakeGitHubRepository>;
+
+        const repositoryName: GitHubRepository = getGitHubRepository(fakeRepository.name);
+        const forkedRepositoryFullName = `${forkedRepositoryOwner}/${repositoryName.name}`;
+        if (contains(fakeRepository.forks, (forkedRepository: FakeGitHubRepository) => forkedRepository.name === forkedRepositoryFullName)) {
```
Comment (kpajdzik) [inline lib/github.ts]: This can be shared with line 650

File: lib/github.ts [hunk 3]
```diff
@@ -651,6 +659,43 @@ export class FakeGitHub implements GitHub {
...
+    const deletedRepository: FakeGitHubRepository | undefined = removeFirst(this.repositories, (repo: FakeGitHubRepository) => repo.name === repositoryFullName);
+
+    let result: Promise<void>;
+    if (!deletedRepository) {
+      result = Promise.reject(new Error(`No fake repository exists with the name "${repositoryFullName}".`));
+    } else {
+      if (deletedRepository.forkOf) {
+        removeFirst(deletedRepository.forkOf.forks, (fork: FakeGitHubRepository) => fork === deletedRepository);
+      }
+      result = Promise.resolve();
```
Comment (kpajdzik) [inline lib/github.ts]: If you return here and @688 you don't need additional variable

File: lib/github.ts (hunk 1 above)
Comment (daschult) [reply lib/github.ts]: I just got in the habit of doing it this way for `getRepository()` that I kept doing it. I don't think there's a good reason. I'll switch it over.

File: lib/github.ts (hunk 3 above)
Comment (daschult) [reply lib/github.ts]: Oh, I know. But I really don't like multiple return statements in a function, so the extra variable is worth it to me.

File: lib/github.ts (hunk 1 above)
Comment (daschult) [reply lib/github.ts]: I just changed a bunch of the `then()` calls to `async/await`.

File: lib/github.ts (hunk 2 above)
Comment (daschult) [reply lib/github.ts]: Done.

**Pull request merged**